packages/config-cli/src/cli/screenshot/screenshot.ts

```diff
@@ -39,104 +45,92 @@ export class CommandScreenShot extends CommandLineAction {
   }
 
   protected onDefineParameters(): void {
+    this.config = this.defineStringParameter({
+      argumentName: 'CONFIG',
+      parameterLongName: '--config',
+      description: 'Path of config bundle file, could be both local file or s3.',
+    });
+
     this.host = this.defineStringParameter({
       argumentName: 'HOST',
       parameterLongName: '--host',
       description: 'Host to use',
-      defaultValue: 'basemaps.linz.govt.nz',
-    });
-
-    this.tag = this.defineStringParameter({
-      argumentName: 'TAG',
-      parameterShortName: '-t',
-      parameterLongName: '--tag',
-      description: 'PR tag(PR-number) or "production"',
-      defaultValue: 'production',
+      defaultValue: DefaultHost,
     });
 
     this.tiles = this.defineStringParameter({
       argumentName: 'TILES',
       parameterLongName: '--tiles',
       description: 'JSON file path for the test tiles',
-      defaultValue: './test-tiles/default.test.tiles.json',
+      defaultValue: DefaultTestTiles,
     });
   }
 
   async onExecute(): Promise<void> {
     const logger = LogConfig.get();
+    const config = this.config.value;
+    let host = this.host.value ?? DefaultHost;
+    const tiles = this.tiles.value ?? DefaultTestTiles;
+
+    let BasemapsServer: FastifyInstance | undefined = undefined;
+    if (config != null) {
+      const port = await getPort({ port: portNumbers(10000, 11000) });
+      host = `http://localhost:${port}`;
+      const server = createServer(logger);
```

Comment:
NIT: why create a new var if we could just use `BasemapsServer`

another option would be `const BasemapsServer = this.startServer(config)` which returns `null | Server`

Reply:
Not sure why if I pass the `BasemapsServer` into the `await new Promise<void>()` if always return type checking errors to say that `BasemapsServer` could be `undefined` even I have checked not null before. So I create a new var to pass it after.

Reply:
yeah if you did this as a function that did `Promise<Server | null>` could be optimised out.